### PR TITLE
feat: add hygen to generate days

### PR DESCRIPTION
**Pull request by danlevings**
As a small thanks, if you're interested 😄

### CODE REVIEW

File: _templates/day/new/day.ejs.t [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  data,
+} = require("./day-<%= day %>");
+
+describe("Day <%= day %>", () => {
+  describe("Part 1", () => {
+    it("works for test data", () => {
+      expect(part1(data("test"))).toBe(undefined);
+    });
+
+    it("works for real data", () => {
```
Comment (evancharlton) [inline _templates/day/new/day.ejs.t:17]: One improvement I've actually made is to run these in the same test suite, like:

```js
// Simple test data
// expect(part1(data([]))).toBe(undefined);

// Test data
expect(part1(data("test"))).toBe(undefined);

// Real data
expect(part1(data))).toBe(undefined);
```

This way during the early phases, I have less test-spam to deal with.

**Review by evancharlton — COMMENTED** [review]
Thanks! I've been meaning to set up `plop` or something but I've deferred for two reasons:
1. I'm lazy.
2. The way I do it allows me to pick up the previous days' enhancements without also having to make changes to the base template.

However, I'm at a pretty stable point in my solution structure, so maybe I'll push this across the line.

Thanks again for doing the legwork!

File: _templates/day/new/day.ejs.t (hunk 1 above)
Comment (danlevings) [reply _templates/day/new/day.ejs.t:17]: That makes sense! i ended up using this cute little run button alot to run only the specific one I wanted
image

Comment (danlevings) [reply _templates/day/new/day.ejs.t:17]: (i should really give that debug button a try sometime...)

Comment (evancharlton) [reply _templates/day/new/day.ejs.t:17]: Protip: open a terminal next to  your IDE and use `yarn --watch` (maybe VS Code can do this; I dunno)

**Comment by evancharlton** [comment]
Thanks for the suggeestion! I took it as inspiration and created a simple "copy this template into the new day" approach (I'm trying to add as few dependencies as possible, just for fun)

**Pull request closed without merging**